### DISCUSSION TITLE
Does debugpy send telemetry?

### DISCUSSION THREAD

**Discussion by PhilippFeO** [microsoft/debugpy#1504] [Q&A] [answered]
Does debugpy send telemetry data? If so, how can I disable it?

Thank you! :)

**Comment by int19h** [comment, marked as answer]
By itself the only thing that debugpy does currently is report its own version to the debug adapter protocol client using the "telemetry" DAP message:
https://github.com/microsoft/debugpy/blob/42853a99c4d3d4d27e4960c8894d780d4f091d68/src/debugpy/adapter/clients.py#L97-L112

Note that what the client does with this telemetry is up to the client, but if they do have their own telemetry, they usually incorporate any such "telemetry" messages into that somehow, and thus you can configure what actually gets submitted in the client. Clients may also choose to report their own telemetry based on other messages in the DAP session, e.g. reporting requests like "stepIn".

We do plan to add more telemetry in the debug adapter and the debug server, but it will all be similarly implemented using "telemetry" DAP events (i.e. no backchannels).

**Reply by PhilippFeO** [reply]
Nice, thank you for the detailed answer!
